### PR TITLE
Update Use Cases in Developer Guide

### LINKED ISSUE #25: Update Use Cases in Developer Guide

### PR DESCRIPTION
**Pull request by yu-sutong**
Close #25

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+**Use case 1: View List of Customers**
 
-**Use case: Delete a person**
+* **System**: SellSavvy
+* **Actor**: USer
```
Comment (FooChao) [inline docs/DeveloperGuide.md:308]: What is USer?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1.  User chooses to view the list of customers.
+2.  SellSavvy retrieves all customers from the database.
+3.  SellSavvy displays all customers along with their details.
+4.  Use case ends.
 
-    Use case ends.
+**Use case 2: View List of Customers**
+
+* **System**: SellSavvy
+* **Actor**: USer
```
Comment (FooChao) [inline docs/DeveloperGuide.md:321]: What is USer?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1a. SellSavvy detects required details missing.
+  * 1a1. SellSavvy displays an error message “Missing details of customer!”.
+  * 1a2. Use case ends.
 
-  Use case ends.
+**Use case 3: Delete Customer and All Orders Related to The Customer**
 
-* 3a. The given index is invalid.
+* **System**: SellSavvy
+* **Actor**: USer
```
Comment (FooChao) [inline docs/DeveloperGuide.md:340]: What is USer?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+4.  User finds the customer index from the list.
+5.  User deletes the customer by his index.
+6.  SellSavvy updates the list of customers and indicates that delete is successful.
+7.  Use case ends.
 
-*{More to be added}*
+**Use case 4: List Orders by Customer**
+
+* **System**: SellSavvy
+* **Actor**: USer
```
Comment (FooChao) [inline docs/DeveloperGuide.md:356]: What is USer?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 3a2. Use case ends.
+
+* 3b. User can list orders made by a customer using their username.
+    * 3b1. User inputs command to list all orders using the username of the intended customer.
+    * 3b2. Use case resumes from step 4.
+
+**Use case 5: Mark Order as Completed**
+
+* **System**: SellSavvy
+* **Actor**: USer
```
Comment (FooChao) [inline docs/DeveloperGuide.md:381]: What is USer?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 1a4. User finds the customer index from the list.
+    * 1a5. User chooses to view all orders from the customer with that customer index.
+    * 1a6. Sellsavvy displays the list of orders under the customer with that customer index.
+    * 1a7. User inputs command to remove marking on the order by the customer index and the order index.
+    * 1a8. Use case resumes to step 8.
+
+**Use case 6: Delete an order**
+
+* **System**: SellSavvy
+* **Actor**: USer
```
Comment (FooChao) [inline docs/DeveloperGuide.md:411]: What is USer?

**Review by FooChao — COMMENTED** [review]
Please changer USer to User.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User chooses to view the list of customers.
+2.  SellSavvy retrieves all customers from the database.
+3.  SellSavvy displays all customers along with their details.
+4.  Use case ends.
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:316]: Consider removing list number from `use case ends` statement
```suggestion
3. Sellsavvy...
Use case ends.
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+**Use case 1: View List of Customers**
 
-**Use case: Delete a person**
+* **System**: SellSavvy
+* **Actor**: USer
+* **Use Case**: UC01 - View List of Customers
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:310]: Consider including guarantees? Based on course use case examples.

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-    Use case ends.
+**Use case 2: View List of Customers**
+
+* **System**: SellSavvy
+* **Actor**: USer
+* **Use Case**: UC02 - Add Customer
+
+**MSS**
+
+1.  User chooses to add a new customer.
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:326]:
```suggestion
1.  User wants to add a new customer.
```

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* **System**: SellSavvy
+* **Actor**: USer
+* **Use Case**: UC03 - Delete Customer and All Orders related to the customer
 
-    * 3a1. AddressBook shows an error message.
+**MSS**
 
-      Use case resumes at step 2.
+1.  User wants to delete a customer.
+2.  User chooses to view all customers.
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:346]: Given link to `view all customers`, would linking to UC01 be better?

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 1a.  Remove marking of order
+    * 1a1. User wants to remove marking on an order due to mistake.
+    * 1a2. User chooses to view all customers.
+    * 1a3. SellSavvy displays the list of customers.
+    * 1a4. User finds the customer index from the list.
+    * 1a5. User chooses to view all orders from the customer with that customer index.
+    * 1a6. Sellsavvy displays the list of orders under the customer with that customer index.
+    * 1a7. User inputs command to remove marking on the order by the customer index and the order index.
+    * 1a8. Use case resumes to step 8.
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:406]: Consider removing list indexing where `use case resumes...`

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-    * 3a1. AddressBook shows an error message.
+**MSS**
 
-      Use case resumes at step 2.
+1.  User wants to delete a customer.
+2.  User chooses to view all customers.
+3.  SellSavvy displays the list of customers.
+4.  User finds the customer index from the list.
+5.  User deletes the customer by his index.
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:349]: `his` may be slightly unclear. Perhaps specifying as `customer index` instead may be clearer?
```suggestion
5.  User deletes the customer by the customer's index in the list.
```

List here would follow from UC01's output list

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* **System**: SellSavvy
+* **Actor**: USer
+* **Use Case**: UC04 - List Orders by Customer
+
+**MSS**
+
+1.  User wants to view all orders made by a specific customer
+2.  User looks at the existing list for the customer name.
+3.  User inputs command to list all orders, by the index of customer in contact list.
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:363]: This is good, I like that source of customer index is clear.

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -302,32 +302,126 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+**Use case 1: View List of Customers**
 
-**Use case: Delete a person**
+* **System**: SellSavvy
+* **Actor**: USer
+* **Use Case**: UC01 - View List of Customers
 
 **MSS**
```
Comment (kaixuan477) [inline docs/DeveloperGuide.md:312]: As per course website use case topics, perhaps removing the whitespace after MSS and following subsections of use cases may be better?

File: docs/DeveloperGuide.md (hunk 13 above)
Comment (yu-sutong) [reply docs/DeveloperGuide.md:363]: Thank you.

**Review by FooChao — APPROVED** [review]
LGTM!

**Pull request merged**